### ISSUE TITLE
Populate fields for match-filter and in the json output

### ISSUE THREAD

**Issue by maaaaz** [yt-dlp/yt-dlp#9835]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm requesting a site-specific feature
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Example URLs

https://www.youtube.com/playlist?list=PLWHarrLXQlqmsKldQC9j9XIMpArq6-0iP

### Provide a description that is worded well enough to be understood

cf. https://github.com/yt-dlp/yt-dlp/issues/9614#issuecomment-2086354473

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
cf. https://github.com/yt-dlp/yt-dlp/issues/9614#issuecomment-2086354473
```

**Comment by maaaaz** [comment]
You can see this comment https://github.com/yt-dlp/yt-dlp/issues/9614#issuecomment-2086354473 that fields in each entry of playlist are not populated.

For public and private videos, fields are "null"

$ yt-dlp "https://www.youtube.com/playlist?list=PLWHarrLXQlqmsKldQC9j9XIMpArq6-0iP" --ignore-errors --flat-playlist --dump-single-json > twiesel_playlist.json

$ cat twiesel_playlist.json | jq
...
        {
            "_type": "url",
            "ie_key": "Youtube",
            "id": "drmZZ9D_W28",
            "url": "https://www.youtube.com/watch?v=drmZZ9D_W28",
            "title": "[Private video]",
            "description": null,
            "duration": null,
            "channel_id": null,
            "channel": null,
            "channel_url": null,
            "uploader": null,
            "uploader_id": null,
            "uploader_url": null,
...

**Comment by maaaaz** [comment]
Yes, it would be a good start, in order to be able to remove private videos from download.

**Comment by bashonly** [comment]
closing in favor of #16665
